torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -782,6 +785,10 @@ def local_tensor(self) -> torch.Tensor:
     @classmethod
     def __torch_function__(cls, func, types, args=(), kwargs=None):
         def dispatch(st: ShardedTensor, func: Callable):
+            # Dispatch to custom user provided op first if it exists.
+            if func in _CUSTOM_SHARDED_OPS:
+                return _CUSTOM_SHARDED_OPS[func](types, args, kwargs, st._process_group)
```

Comment:
what's the relationship between a custom sharded op and a custom sharding_spec op? What if user passed in a custom sharding spec and the custom sharding specs have some ops in there, should a custom sharded op overriding the custom sharding spec op?

My guess is we should override, and this means user facing level, we have two(order matters): 1. custom sharded op 2. custom sharding spec op, we should probably document these two type of registrations to the user. 

internally we have: 1. custom sharded op 2. custom sharding spec op 3. default sharded op (order matters).

Reply:
Does this mean we are going to allow users to use their own customized op? Is there already a use case here?

Reply:
> Does this mean we are going to allow users to use their own customized op? Is there already a use case here?

Yes, this was actually the reason to add `sharded_op_impl` decorator.

> internally we have: 1. custom sharded op 2. custom sharding spec op 3. default sharded op (order matters).

Yeah, let me add documentation that this is the order in which we allow the custom ops.